libr/util/str.c

```diff
@@ -1505,8 +1506,142 @@ R_API char *r_str_escape_utf32be(const char *buf, int buf_size, bool show_asciid
 	return r_str_escape_utf (buf, buf_size, R_STRING_ENC_UTF32BE, show_asciidot, esc_bslash, false);
 }
 
-// JSON has special escaping requirements
-// TODO: merge with r_str_escape_utf() and r_str_byte_escape() using RStrEsc
+R_API char *r_str_encoded_json(const char *buf, int buf_size, const char *encoding) {
```

Comment:
`encoding` should be an enum. You can make a helper function to quickly get the enum from the `r_config_get (r->config, "json.encoding")` string.

Reply:
agree. PJ_ENCODING_BASE64 and so on

Reply:
also add PJ_ENCODING_DEFAULT